modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -4934,7 +4938,7 @@ public CodegenParameter fromRequestBody(RequestBody body, Set<String> imports, S
                 codegenParameter.isModel = true;
             }
 
-            if (codegenModel != null && !codegenModel.emptyVars) {
+            if (codegenModel != null && (codegenModel.hasVars || useAliasReferenceName)) {
```

Comment:
@bodograumann why route the aliased models to these two clauses?
* isObjectschema
* Codegenmodel != null
Per our discussion doesn't this only apply to array and map aliases?
Why not handle those cases in those clausee above only?

Reply:
The clauses for array and map aliases above are special cases where the model is not referenced by name, but included as an array or object literal.
With `generateAliasAsModel` we disable this special handling, so they should return to where they would normally be handled if they weren’t aliases. @spacether

Reply:
@bodograumann I am having trouble understanding. Can you paste in a spec example here or in the PR description?

Reply:
I have added an example to the description @spacether.

Reply:
Thank you for adding that. It has helped me understand this use case.